Implement top navigation tabs

borntojesus opened the pull request:
- Create top navigation tabs layout
- Add mobile responsive design
- Implement box shadow for better visibility
- Adjust spacing and sizing for mobile view

## Preview Link

https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/

## Checklist

- [ ] Added a **preview link**
- [ ] Reviewed AI PR Agent suggestions
- [ ] **For Tyk Members** - Added Jira DX PR ticket to the subject
- [ ] **For Tyk Members** - Added the appropriate release labels (for fixes add the latest release)

## New Contributors 
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.

letzya reviewed (changes requested):
GReat PR @borntojesus , thank you!
1. The home page needs to look like in v5.5 https://tyk.io/docs/5.5/contribute/
2. There's no need for a line above the first label 
4. I will create a pr with the menu.yaml changes so we can see how it looks
5. The first tab should be "homepage", as in 5.5. No navigation. 
6. The "overview" tab content should be in the API management tab.
7. We now have content for 2 tabs: "AI management" and "API governance"
8. Can I please ask about the implementation choice to add "tab_on_top: true" and not add a new category "category: tab_on_top"

borntojesus commented:
Updated the PR with your feedback, please have a look

scott-tyk-web commented on tyk-docs/static/js/docs-table-of-contents.js:
When visiting this url
https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/tyk-cloud/#using-api-blueprint

The wrong item in the right menu is highlighted. This is probably to do with the new extra header (the detected item is likely behind the header.)

You might have to adjust this line:
const headerHeight = 30; // Adjust this value to match the height of your fixed header

in the getHighestHeader() function to compensate for this.

Otherwise this seems good.

scott-tyk-web reviewed (commented):
I am not that familiar with the markdown side of this site to comment on that really. The CSS and the Javascript look ok apart from one comment I have made.

When testing this the AI Studio link does not work though?

Also, on some smaller mobile screens the tabs are extend off screen. If other tabs are added in the future they could be missed? Maybe this needs to be another hamburger.

borntojesus commented:
thanks Scott, will take a look

borntojesus commented:
@letzya @scott-tyk-web I added a value for headerHeight, and everything works as expected. The mobile tabs have horizontal scrolling, and they display correctly on smaller screens.

However, I noticed that the Netlify preview branch doesn’t seem to be updated — I made changes to the homepage and added a new menu, but these updates don’t appear on this link [this link](https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/). It’s possible that Netlify didn’t pick up the latest commit (maybe due to caching). 

Could you please check if the latest changes were deployed correctly?

sharadregoti commented:
> I made changes to the homepage and added a new menu, but these updates don’t appear on this link [this link](https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/). It’s possible that Netlify didn’t pick up the latest commit (maybe due to caching).

Hey @borntojesus, since you are not part of the Tyk Organization, Netlify preview must be manually approved for each commit you make. I have approved your request; the change will be live on the preview URL shortly.

sharadregoti commented:
**Observations:**
1. The version selector dropdown in the left-side navigation is not visible unless I scroll to the bottom of the page. On the live site, it is visible on every page.
![image](https://github.com/user-attachments/assets/b98734b3-0a2f-432d-ab76-dca51c866227)
2. Alignment issues
![Screenshot 2025-05-02 at 5 35 18 PM](https://github.com/user-attachments/assets/ce7dcf14-c6b2-4801-b030-0928a08a9b7a)

letzya commented:
@borntojesus our repo keeps changing so a reminder to update it from master. thanks

borntojesus commented:
I've made some fixes. Could you please take a look and let me know if everything is okay?

letzya commented:
@borntojesus do you know why you have missing paths 
https://github.com/TykTechnologies/tyk-docs/actions/runs/14797846683/job/41553300174?pr=6354

letzya commented on tyk-docs/data/menu.yaml:
```diff
@@ -25,31 +28,6 @@ menu:
-        category: Page
-        show: True
-      - title: "Configure Your First API"
-        path: /getting-started/configure-first-api
```
Why have you removed this url? (and 2 more?)

letzya commented on tyk-docs/data/menu.yaml:
```diff
@@ -79,274 +57,320 @@ menu:
-        show: True
   - title: "Developer Portal"
-    category: Label
+    category: Tab
```
Why do we have 2 definitions of categories for a tab?

borntojesus replied:
I removed the old tab_on_top values and made the logic using categories

letzya replied:
Gosh, sorry, my bad! thanks

sharadregoti commented:
**Observations**
1. The left navigation goes blank when visiting any nested [page](https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/ai-management/ai-studio/quickstart/).
![image](https://github.com/user-attachments/assets/b2e5cc2b-7692-437b-8d43-da60f2bc9769)
2. In my opinion, this should either be aligned with the left navigation or the top margin should be slightly reduced.
![Screenshot 2025-05-05 at 1 21 13 PM](https://github.com/user-attachments/assets/95bf0f7b-7a52-4bd0-b7bf-7d138f6e5da5)

borntojesus commented:
Working on the fixes — thanks for the feedback!

borntojesus commented:
@letzya When you have a moment, could you please manually approve the CI for this PR?

letzya commented:
@borntojesus 
1. Remove duplication of tabs (no need to put the tab in the right nav)
2. A label must always have a line above, except if there's a label in the first line of the right nav (we can have regular page category entries, and then, the first label might come further down below)

borntojesus commented:
one more bug
<img width="2936" alt="image" src="https://github.com/user-attachments/assets/ad4396ee-6d35-473e-b01b-a3d420124873" />

borntojesus commented:
> @borntojesus
> 
> * [ ]  1. Remove duplication of tabs (no need to put the tab in the left nav)
> * [ ]  2. A label must always have a line above, except if there's a label in the first line of the right nav (we can have regular page category entries, and then, the first label might come further down below)
> * [ ]  3. Put developer support back in API management. Make sure to change/add only tabs, not the rest (where possible)
> * [ ]  4. h4 in the right nav is not showing in full (the upper bar of the tabs is hidding it)

borntojesus commented:
> @borntojesus
> 
> * [ ]  1. Remove duplication of tabs (no need to put the tab in the left nav)
> * [ ]  2. A label must always have a line above, except if there's a label in the first line of the right nav (we can have regular page category entries, and then, the first label might come further down below)
> * [ ]  3. Put developer support back in API management. Make sure to change/add only tabs, not the rest (where possible)
> * [ ]  4. h4 in the right nav is not showing in full (the upper bar of the tabs is hidding it)

borntojesus commented:
All requested changes have been addressed:

1. Removed duplication of tabs from the left nav
2. Labels now always have a line above, following the rules mentioned
3. Developer support is added back to API Management
4. Fixed the h4 overlap in the right nav — it's now fully visible

Fixed an issue where the sticky banner was overlapping the top tabs — added spacing and/or layout adjustments to prevent this 

Also, I accidentally closed the PR briefly — reopened it right after. Let me know if anything else needs attention!

letzya commented on tyk-docs/data/menu.yaml:
```diff
@@ -3,7 +3,7 @@
   - title: "Home"
     path: /
-    category: Label
+    category: Tab
```
Where is this section in the navigation https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/tyk-components/ @borntojesus

letzya commented:
> lso, I accidentally closed the PR briefly — reopened it right after. Let me know if anything else needs attention!

Need to see how it looks with the menu.yaml I have given you.

letzya commented on tyk-docs/themes/tykio/layouts/shortcodes/tabs_end.html:
```diff
@@ -37,7 +37,8 @@
     var navTabs = $(this.parentNode.parentNode).find('> .nav-tabs'),
         title = $(this).attr('title');
-    navTabs.append('<li><a href="#">'+title+'</a></li');
+    var anchor = title.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/(^-|-$)/g, '');
```
Thanks for this. Did you use this shortcode somewhere in this pr?

borntojesus replied:
I've reverted to the previous version of the shortcode for now. Let me know if you'd like me to implement the suggested anchor format (subheading + tab name) as a follow-up.

sharadregoti commented:
**Observations**
1. Our header navigation jumps when a banner is displayed.
I believe the live website has a sticky header, with the banner overlaying it.

https://github.com/user-attachments/assets/de884b58-3c2c-43af-85fe-24189089f74e

https://github.com/user-attachments/assets/db52391e-e08e-4aa8-897e-452ca6461c55

2. When you click on any item in the right-side navigation, it takes you to that section, but you can still see the content from the previous section. This was not the case on the live website.
https://deploy-preview-6354--tyk-docs.netlify.app/docs/nightly/api-management/upstream-authentication/#request-signing
<img width="1512" alt="Screenshot 2025-05-08 at 12 41 59 PM" src="https://github.com/user-attachments/assets/1b5f8769-e794-4c3d-a5ef-df755870e0cf" />

letzya commented:
@borntojesus looks good!
2 small issues
1. When there's a banner, it's hiding the bottom of the search box under the tabs. Can we do something about it? (Hiding the top of the search would make more sense. Also, check version tyk.io/5.5/ - there's a bar and it's not hiding anything, it just moves. Why is that?
2. While scrolling up, it slightly scrollsand  moves the search bar under the tabs. Is it the same problem as #1?

borntojesus commented:
@letzya @sharadregoti I have fixed the following issues:
- The navigation and search bar are no longer overlapped by the banner, even when it appears.
- The search bar and tabs no longer shift or move under the banner while scrolling.
- Simplify tab structure and remove fixed positioning
- Update mobile styles for better responsiveness

Please let me know if you notice any further issues!

sharadregoti commented:
/probe "Can you review this PR?"

letzya commented:
> /probe "Can you review this PR?"

Might not work since it's an external branch. Not sure why though @sharadregoti
